packages/web-components/src/components/toggle/toggle.ts

```diff
@@ -138,32 +184,46 @@ class CDSToggle extends HostListenerMixin(CDSCheckbox) {
 
     const labelTextClasses = classMap({
       [`${prefix}--toggle__label-text`]: labelText,
+      [`${prefix}--visually-hidden`]: hideLabel,
     });
-    const stateText = checked ? labelA : labelB;
+
+    let stateText = '';
+
+    if (hideLabel) {
+      stateText = labelText || '';
+    } else {
+      stateText = checked ? labelA : labelB;
+    }
+
+    const labelId = id ? `${id}_label` : undefined;
+
+    const hasLabelText = (this.labelText ?? '') !== '';
+
+    const ariaLabelledby =
+      this.ariaLabelledby ?? (hasLabelText && labelId ? labelId : undefined);
```

Comment:
```suggestion
      this.ariaLabelledby ?? (hasLabelText && labelId);
```

I think this is the does the same?

Reply:
Yes, it'll have the same function thanks for pointing that out